### ISSUE TITLE
WandbLogger `save_dir` and `dir` parameters do not work as expected.  

### ISSUE THREAD

**Issue by Jigar1201** [Lightning-AI/pytorch-lightning#19830]
### Bug description

The `save_dir` param is set to `.` by default and hence `if save_dir is not None:` is always False.
https://github.com/Lightning-AI/pytorch-lightning/blob/d1949766f8cddd424e2fac3a68b275bebe13d3e4/src/lightning/pytorch/loggers/wandb.py#L327C12-L327C20

When a user only specifies the `dir` and does specify `save_dir`, the current implementation ignores `dir` because `save_dir` is not None (set to `.` by default). The priority should be 1. Use `save_dir` if provided, 2. Use dir if `save_dir` is not present. 3. Use `.` which is the default. 
If this is not a duplicate of an issue and the workflow is expected as described above, let me know and I will be interested in creating a PR to fix this. 

### What version are you seeing the problem on?

v2.1

### How to reproduce the bug

```python
wandb_logger = WandbLogger(
        project="PROJECT", dir=ignored_dir, id=timestr, name=config.run_id, resume=False, job_type="testing", log_model="all")
```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by angerhang** [comment]
> When a user only specifies the `dir` and does specify `save_dir`, the current implementation ignores `dir` because `save_dir` is not None (set to `.` by default). The priority should be 1. Use `save_dir` if provided, 2. Use dir if `save_dir` is not present. 3. Use `.` which is the default.

Yeah the current difference between `dir` and `save_dir` is confusing. In the documentation this difference is not explained in terms of priority. Perhaps, we can elaborate on expected behaviour when setting `dir` and `save_dir` individually or when both of them are set.

**Comment by mukeshr06** [comment]
I reproduced this on current master: passing dir alone is still overridden by the default save_dir. I am preparing an AI-assisted fix that resolves the aliases before applying the default, documents save_dir precedence, and adds regression tests for strings, Path objects, and explicit save_dir=".". The focused logger suite passes locally (24 tests).
